### ISSUE TITLE
Disable sending emails when a package is published

### ISSUE THREAD

**Issue by bradgessler** [npm/registry-issue-archive#284]
Every time an NPM package is published, an email gets sent that I don't want. Here's what that looks like:

```
Hi acme!

A new version of the package @acme/toolkit (5.9.4-elements-rc1) was published at 2018-03-01T19:50:26.426Z from
13.28.25.216. The shasum of this package was 5142578bdbf583276f7ab62d4b1c1db041e9e151.

If you have questions or security concerns, you can reply to this message or
email support@npmjs.com.

npm loves you.
```

This email is very disruptive when updates are frequently published to many packages. I recommend a few remedies:

1. Give users the option to disable this notification (preferred)
2. Remove "npm loves you", because if you did, support@ and isaac@ wouldn't ignore my requests for the change above. If you loved me you would be much more empathetic to my situation.
3. Revert the behavior to not send emails at all.

**Comment by hfreire** [comment]
Hey guys! 

As @bradgessler said this is quite disruptive/spammy when handling a significant number of packages and a high release frequency environment.

Can we allow the user to disable this feature?

PS: Can we instead integrate this notification with Slack?

**Comment by andrewryan1906** [comment]
Definitely would like to +1 this... it's really annoying.

**Comment by cstoffer** [comment]
Same here! +1

**Comment by kujon** [comment]
+1

**Comment by vsych** [comment]
+1

**Comment by andyklaiber** [comment]
+1

**Comment by rubenhak** [comment]
too much love in my life. +1 to allow disabling email notifications.

**Comment by drazvan** [comment]
+1

**Comment by JRGranell** [comment]
+1

**Comment by kujon** [comment]
A new version of the comment +1 https://github.com/npm/registry/issues/284#issuecomment-389110849 was published.

**Comment by pedronauck** [comment]
please, give us this 🙏  it's reaaaaaally annoying 😞

**Comment by heldr** [comment]
+1 we definitely need this option 😞

**Comment by jaimemarijke** [comment]
+1

**Comment by guodong** [comment]
+1

**Comment by kzantow** [comment]
+1

**Comment by rapidoo** [comment]
+1

**Comment by ghost** [comment]
For those of you working with mono-repositories, I've filed a separate issue to request publishes that happen in quick success to be bundled into a single e-mail: #367

**Comment by JonnyPickard** [comment]
+1

**Comment by Fjandin** [comment]
+1

**Comment by intcreator** [comment]
Usually when this happens I end up filtering the emails/marking them as spam since I want control over my inbox.  Even if that means possibly missing an important email or alerting Google's spam filters that the sender can't be trusted.

**Comment by verekia** [comment]
I'll weight in saying that it is particularly disruptive when the email address used is a  general company address that broadcasts to many employees. Really really spammy. I don't think creating an email address just for NPM spam, or using one employee's address for the company's NPM use are reasonable workarounds.

**Comment by pedrotorchio** [comment]
+2

**Comment by lachrist** [comment]
+1

**Comment by BuGlessRB** [comment]
I give npm exactly 48 hours to solve this.  After that I'll start reporting their domain and mails to all spamcop etc. triggers I can find.

**Comment by verekia** [comment]
Geez, is this https://www.reddit.com/r/iamverybadass ?

**Comment by BuGlessRB** [comment]
Given the fact that it already seems to be in effect since at least March 1st, it is ample time to fix a spam problem.  Reporting their mails to spamcop et al solves two problems: it mitigates the spam, and creates another awareness level for support.

**Comment by verekia** [comment]
Even if you could do anything about that, getting their domain flagged as spam would be damaging to our entire community for the emails that matter. So just, no.

**Comment by BuGlessRB** [comment]
npm is not living up to their own mission statement:

> npm’s mission is to take Open Source development to entirely new places. When everyone else is adding force, we work to reduce friction.
> npm is not a typical product, and we are not a typical “work hard/play hard” startup. We are responsible adults with diverse backgrounds and interests, who take our careers and our lives seriously. We believe that the best way to iterate towards success is by taking care of ourselves, our families, our users, and one another. We aim for a sustainable approach to work and life, because that is the best way to maximize long-term speed, while retaining clarity of vision. Compassion is our strategy.

I presume npm is a company with paid-full-time-professionals.  Then why don't they act like that?

**Comment by verekia** [comment]
I don't think you even read that statement that you quote. It literally says that they are taking it easy to be more sustainable. Stop trying to make noise, add your +1 emoji like everyone else, and let them prioritize their issues the way they think is right.

Yes, these emails are annoying, but that is still a mild annoyance. They surely have more critical things to deal with, and you should probably not let that upset you this much.

**Comment by BuGlessRB** [comment]
Well, given the fact that I can still remember what a spam-free internet looked like, I may have different priorities.  Sustainability includes not wasting energy, and not wasting energy implies not sending out meaningless emails on a large scale (i.e. spam).  I've been fighting spam since the first spam-message appeared; and history has taught me that unless you take a firm stance, spam tends to get out of hand quickly.

In any case, considering your pleas, I'll refrain from actively reporting their domain anywhere, but I will start marking the messages as spam (as undoubtedly others are already doing) in my mailbox, which eventually results in backpressure to the global spamreporting network (in the various automated mechanisms already in place).  It would be a lot saner, if npm offered to opt out of these emails instead.

**Comment by heldr** [comment]
🤔 that being said, should we take the noise you've caused in this issue as spam?

**Comment by markcellus** [comment]
I dont know, but I'd much prefer if the future replies on this thread be only productive and for everyone to put their + ones on the original post instead of making an entirely new reply for it

**Comment by bradgessler** [comment]
I’d recommend reporting unsolicited emails to https://consumercomplaints.fcc.gov/hc/en-us if you’ve given up on NPM fixing this and don’t want to trigger spam filters.

That said, it’s reasonable to assume NPM emails will get marked as spam and their delivery success rates will decrease since they are being slow to fix the problem. Unfortunately there’s nothing any of us can do about it because there’s way more people who have not read this issue who have quietly marked these messages as spam.

That said, please keep the responses of this post focused on what NPM should do to fix the issue and not on what we, the users should do, because frankly this issue is not our fault and outside of our control. The ball is in NPM’s court.
